Allow disabling app services auth (#1932)
Azure-Samples/azure-search-openai-demo: docs/login_and_acl.md

In both the chat and ask a question modes, under **Developer settings** optional **Use oid security filter** and **Use groups security filter** checkboxes will appear. The oid (User ID) filter maps to the `oids` field in the search index and the groups (Group ID) filter maps to the `groups` field in the search index. If `AZURE_ENFORCE_ACCESS_CONTROL` has been set, then both the **Use oid security filter** and **Use groups security filter** options are always enabled and cannot be disabled.

#### Programmatic Access with Authentication

> If you want to use the chat endpoint without the UI and still use authentication, you must disable the integrated app services authentication and use the authentication built into the sample app instead. Ensure the `AZURE_DISABLE_APP_SERVICES_AUTHENTICATION` environment variable is set before deploying.

Reviewer: ```suggestion
If you want to use the chat endpoint without the UI and still use authentication, you must disable [App Service built-in authentication](https://learn.microsoft.com/azure/app-service/overview-authentication-authorization) and use only the app's MSAL-based authentication flow. Ensure the `AZURE_DISABLE_APP_SERVICES_AUTHENTICATION` environment variable is set before deploying.
```

Get an access token that can be used for calling the chat API using the following code:

> ```

Reviewer: ```suggestion
```python
```

from azure.identity import DefaultAzureCredential
import os

> token = DefaultAzureCredential().get_token(f"api://{os.environ['AZURE_SERVER_APP_ID']}/access_as_user", tenant_id=os.getenv('AZURE_AUTH_TENANT_ID', os.getenv('AZURE_TENANT_ID')))

Reviewer: This code only works in prod, since AZURE_SERVER_APP_ID isnt exported locally. Is that intentional? If so, you should specify that this needs to be run in the App Service SSH.

Reviewer: Nevermind! I forgot that we set that ourself, in the scripts.

- `AZURE_ENFORCE_ACCESS_CONTROL`: Enforces Entra ID based login and document level access control on documents with access control assigned. Set to true before running `azd up`. If `AZURE_ENFORCE_ACCESS_CONTROL` is enabled and `AZURE_ENABLE_UNAUTHENTICATED_ACCESS` is not enabled, then authentication is required to use the app.
- `AZURE_ENABLE_GLOBAL_DOCUMENT_ACCESS`: Allows users to search on documents that have no access controls assigned
- `AZURE_ENABLE_UNAUTHENTICATED_ACCESS`: Allows unauthenticated users to access the chat app, even when `AZURE_ENFORCE_ACCESS_CONTROL` is enabled. `AZURE_ENABLE_GLOBAL_DOCUMENT_ACCESS` should be set to true to allow unauthenticated users to search on documents that have no access control assigned. Unauthenticated users cannot search on documents with access control assigned.

> - `AZURE_DISABLE_APP_SERVICES_AUTHENTICATION`: Disables [use of the integrated authentication for App Services](https://learn.microsoft.com/azure/app-service/overview-authentication-authorization). Built-in authentication to the chat app is used instead. Useful when you want to provide programmatic access to the chat endpoints with authentication.

Reviewer: ```suggestion
- `AZURE_DISABLE_APP_SERVICES_AUTHENTICATION`: Disables [use of built-in authentication for App Services](https://learn.microsoft.com/azure/app-service/overview-authentication-authorization). An authentication flow based on the MSAL SDKs is used instead. Useful when you want to provide programmatic access to the chat endpoints with authentication.
```

Reviewer: The docs usually call it "built-in" so am avoiding calling our MSAL flow "built in".
